src/powerbox/tools.py

```diff
@@ -325,7 +550,10 @@ def angular_average_nd(
 
     if len(coords) != len(field.shape):
         raise ValueError("coords should be a list of arrays, one for each dimension.")
-
+    if interpolation_method is not None and angular_resolution is None:
+        angular_resolution = 1.0 / 2 ** (len(coords) - n)
```

Comment:
I think the default angular resolution should be such that the inter-spherical-grid spacing in the _outer_ shell roughly matches the coordinate spacing of the rectangular grid, if that makes sense

Reply:
That does make sense. Currently it does not though. It's not straightforward to me how to initialize the number of samples for each angle and radius such that they roughly match sumweights for different numbers of dimensions to average over and different HII_DIMs. I think a reasonable short term solution is to maybe add a warning that whatever we leave as default may not suit everyone. It's not difficult to find the right values for one use case (e.g. 3D chunk to 1 or 2D PS) but it's difficult to find a formula that generalises. So users would just make their own based on what they want to do. I can take more time to do this properly later. For now I'll just leave in something simple. It does not impact the results very much in either case for our applications (since at larger k the variance is very low so there is no need to sample all of the points anyways).

Reply:
Also the whole issue would be solved if we could just add sumweights as a potential input to the generator. But that's a bit ugly.